lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
-            value[:headers].each do |header|
-              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+      if (!headers.nil? && headers.any?) || !grandchildren.empty?
+        if has_child_pages
+          active = filename.active?(url, exact: true)
+          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
+          sub_group_markup << "<li class=\"#{item_class}\">"
+          sub_group_markup << '<div class="nav-subgroup-heading">'
+          sub_group_markup << '<i class="material-icons">arrow_right</i>'
+          sub_group_markup << "<a href=\"#{url}\">#{title}</a>"
+          sub_group_markup << '</div>'
+          sub_group_markup << '<ul class="nav-ul">'
+
+          if grandchildren.empty?
+            headers.each do |header|
+              hash = header[:hash]
+              subtitle = header[:title]
+              sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
             end
           else
-            subgroup_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
-            subgroup << "<li class=\"#{subgroup_leaf_class}\"><a href=\"#{value[:url]}\">#{title} overview</a></li>"
+            sub_group_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
+            sub_group_markup << "<li class=\"#{sub_group_leaf_class}\"><a href=\"#{url}\">#{title} overview</a></li>"
 
-            subsubgroup_list.each do |subsubgroup_key, subsubgroup_value|
-              subgroup << generateSubgroup(filename, subsubgroup_key, subsubgroup_value, subsubgroup_list, 3)
+            grandchildren.each do |grandchild_path, grandchild_page|
+              sub_group_markup << generate_sub_group(filename, grandchild_path, grandchild_page, grandchildren, 3)
             end
           end
 
-          subgroup << '</ul>'
-          subgroup << '</li>'
+          sub_group_markup << '</ul>'
+          sub_group_markup << '</li>'
         else
-          value[:headers].each do |header|
-            subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+          headers.each do |header|
+            hash = header[:hash]
+            subtitle = header[:title]
+            sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
           end
         end
       else
-        subgroup << if has_subgroups
-                      "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    else
-                      "<li class=\"nav-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    end
+        sub_group_markup << if has_child_pages
+                              "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{url}\">#{title}</a></li>"
```

Comment:
Codacy found an issue: [Line is too long. [109/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)